lib/context_adapter_helper.js

```diff
@@ -25,48 +25,47 @@
 
 var caLogger = require('logops'),
     caConfig = require('./context_adapter_configuration'),
-    caPackage = require('../package.json');
+    caPackage = require('../package.json'),
+    uuid = require('uuid');
 
 /**
- * Generates a 32 bit integer hash code
- * @param {string} str The seed
- * @return {number} The hash code
+ * Returns the correlator included in a request, if any
+ * @param {object} data It can be an HTTP request or an array of HTTP headers
+ * @return {string} The correlator, if any
  */
-function getHashCode(str) {
-  var hash = 0, i, chr, len;
-  if (str && str.length === 0) {
-    return hash;
-  }
-  for (i = 0, len = str.length; i < len; i++) {
-    chr = str.charCodeAt(i);
-    hash = ((hash << 5) - hash) + chr;
-    hash |= 0; // Convert to 32bit integer
-  }
-  return hash;
+function getCorrelator(data) {
```

Comment:
If no correlator is found in headers al the logging messages will appear with `corr=n/a' a new internal transaction id will be generated per request and all the related logging messages will show it in the 'trans' context property.